src/test/java/org/jabref/gui/duplicationFinder/DuplicateSearchTest.java

```diff
@@ -0,0 +1,112 @@
+        if (Globals.prefs == null) {
+            Globals.prefs = JabRefPreferences.getInstance();
+        }
+
+        entry1 = new BibEntry(StandardEntryType.InProceedings)
+                .withField(StandardField.AUTHOR, "Souti Chattopadhyay and Nicholas Nelson and Audrey Au and Natalia Morales and Christopher Sanchez and Rahul Pandita and Anita Sarma")
+                .withField(StandardField.TITLE, "A tale from the trenches")
+                .withField(StandardField.YEAR, "2020")
+                .withField(StandardField.DOI, "10.1145/3377811.3380330")
+                .withField(StandardField.SUBTITLE, "cognitive biases and software development")
+                .withCitationKey("Chattopadhyay2020");
+
+        when(jabRefFrame.getCurrentLibraryTab()).thenReturn(libraryTab);
+        when(stateManager.activeDatabaseProperty()).thenReturn(OptionalObjectProperty.empty());
+        duplicateSearch = new DuplicateSearch(jabRefFrame, dialogService, stateManager);
+    }
+
+    @Test
+    public void executeWithNoEntries() {
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.of(bibDatabaseContext));
+        when(bibDatabaseContext.getEntries()).thenReturn(Collections.emptyList());
+
+        duplicateSearch.execute();
+        verify(dialogService, times(1)).notify(Localization.lang("Searching for duplicates..."));
+    }
+
+    @Test
+    public void executeWithOneEntry() {
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.of(bibDatabaseContext));
+        when(bibDatabaseContext.getEntries()).thenReturn(Collections.singletonList(entry1));
+
+        duplicateSearch.execute();
+        verify(dialogService, times(1)).notify(Localization.lang("Searching for duplicates..."));
+    }
+
+    @Test
+    public void executeWithNoDuplicates() {
+        BibEntry entry2 = new BibEntry(StandardEntryType.InProceedings)
+                .withField(StandardField.AUTHOR, "Tale S Sastad and Karl Thomas Hjelmervik")
+                .withField(StandardField.TITLE, "Synthesizing Realistic, High-Resolution Anti-Submarine Sonar Data\n")
+                .withField(StandardField.YEAR, "2018")
+                .withField(StandardField.DOI, "10.1109/OCEANSKOBE.2018.8558837")
+                .withCitationKey("Sastad2018");
+
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.of(bibDatabaseContext));
+        when(bibDatabaseContext.getEntries()).thenReturn(Arrays.asList(entry1, entry2));
+        when(bibDatabaseContext.getMode()).thenReturn(BibDatabaseMode.BIBTEX);
+        when(libraryTab.getBibDatabaseContext()).thenReturn(bibDatabaseContext);
+        when(libraryTab.getDatabase()).thenReturn(mock(BibDatabase.class));
+        when(libraryTab.getUndoManager()).thenReturn(undoManager);
+        when(undoManager.addEdit(mock(NamedCompound.class))).thenReturn(true);
+
+        duplicateSearch.execute();
+        verify(dialogService, times(1)).notify(Localization.lang("Searching for duplicates..."));
```

Comment:
Hi :)
I saw on the discussion #7636 that you would prefer using any() over a localized argument for the notify() method. The rationale for using this localized argument was to verify if dialogService shows the correct message for each condition. As @ningxie1991 pointed out, would the dialogService somehow loose its purpose when using any() as the argument?

Moreover, in this case, if I use the localized argument, both very() methods work, but by using any() as argument I receive the feedback that the dialogService got invoked only once. So, what did I miss on here?

Reply:
It's okay to use the correct localization argument. I think we thought that it just verifies the numbers of calls, but not the content